### PR TITLE
[Java] Fix `setCurrentTask()` in multi threading

### PR DESCRIPTION
**Pull request by jovany-wang**
## What do these changes do?
For actor tasks , we should not set current driver id to null.

## Related issue number
N/A

### CODE REVIEW

File: java/runtime/src/main/java/org/ray/runtime/Worker.java [hunk 1]
```diff
@@ -95,7 +96,15 @@ public void execute(TaskSpec spec) {
...
+      if (spec.isActorTask() || spec.isActorCreationTask()) {
+        // Don't need to reset current driver id if the worker is an actor.
+        // Because the following tasks should all have the same driver id.
+        runtime.getWorkerContext().setCurrentTask(null, spec.driverId, null);
+      } else {
+        // For normal task, we should set current task id to null as well.
+        runtime.getWorkerContext().setCurrentTask(null, null, null);
+      }
+
       Thread.currentThread().setContextClassLoader(oldLoader);
```
Comment (stephanie-wang) [inline java/runtime/src/main/java/org/ray/runtime/Worker.java]: Sorry I'm not clear on what `classLoader` is, but it seems like you could get rid of this line and pass in `oldLoader` in the `setCurrentTask` lines above?

File: java/runtime/src/main/java/org/ray/runtime/WorkerContext.java [hunk 2]
```diff
@@ -63,18 +63,15 @@ public UniqueId getCurrentTaskId() {
    * Set the current task which is being executed by the current worker. Note, this method can only
    * be called from the main thread.
    */
-  public void setCurrentTask(TaskSpec task, ClassLoader classLoader) {
+  public void setCurrentTask(UniqueId currentTaskId,
```
Comment (stephanie-wang) [inline java/runtime/src/main/java/org/ray/runtime/WorkerContext.java]: Set `currentTaskId` to `UniqueId.NIL` if it's `null`?

**Review by stephanie-wang — COMMENTED** [review]
Thanks! Left a couple questions.

File: java/runtime/src/main/java/org/ray/runtime/WorkerContext.java (hunk 2 above)
Comment (raulchen) [reply java/runtime/src/main/java/org/ray/runtime/WorkerContext.java]: right, but maybe it's more clear to just pass in `UniqueId.NIL`

File: java/runtime/src/main/java/org/ray/runtime/Worker.java (hunk 1 above)
Comment (raulchen) [reply java/runtime/src/main/java/org/ray/runtime/Worker.java]: @stephanie-wang `classLoader` is used to tell JVM where and how to load a class into memory. We use different class loaders for different drivers. So different drivers can have classes with the same name without conflicts. 
@jovany-wang `classLoader` should be set and reset in the same way as the driver id.

**Comment by jovany-wang** [comment]
@raulchen  @stephanie-wang 
I have refined this PR with following logic:
In `finally` block we should not set any workerContext.
Because we will set it in the next turn of the while loop.

**Review by raulchen — APPROVED** [review]

**Pull request merged**